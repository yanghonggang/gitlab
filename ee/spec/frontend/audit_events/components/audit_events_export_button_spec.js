import { GlButton } from '@gitlab/ui';
import { shallowMount } from '@vue/test-utils';

import AuditEventsExportButton from 'ee/audit_events/components/audit_events_export_button.vue';

const EXPORT_HREF = 'http://example.com/audit_log_reports.csv?created_after=2020-12-12';

describe('AuditEventsExportButton component', () => {
  let wrapper;

  const findExportButton = () => wrapper.find(GlButton);

  const createComponent = (props = {}) => {
    return shallowMount(AuditEventsExportButton, {
      propsData: {
        exportHref: EXPORT_HREF,
        ...props,
      },
    });
  };

  beforeEach(() => {
    wrapper = createComponent();
  });

  afterEach(() => {
    wrapper.destroy();
  });

  describe('Audit Events CSV export button', () => {
    it('matches the snapshot', () => {
      expect(wrapper.element).toMatchSnapshot();
    });

    it('renders the Audit Events CSV export button', () => {
      expect(findExportButton().exists()).toBe(true);
    });

    it('renders the export icon', () => {
      expect(findExportButton().props('icon')).toBe('export');
    });

    it('links to the CSV download path', () => {
      expect(findExportButton().attributes('href')).toEqual(EXPORT_HREF);
    });
  });
});
