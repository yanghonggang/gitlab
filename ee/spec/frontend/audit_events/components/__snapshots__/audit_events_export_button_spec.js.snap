// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuditEventsExportButton component Audit Events CSV export button matches the snapshot 1`] = `
<gl-button-stub
  buttontextclasses=""
  category="primary"
  href="http://example.com/audit_log_reports.csv?created_after=2020-12-12"
  icon="export"
  size="medium"
  title="Max 100,000 events"
  variant="default"
>
  
  Export as CSV

</gl-button-stub>
`;
